src/mongocxx/test/spec/unified_tests/runner.cpp

```diff
@@ -284,27 +284,28 @@ void add_ignore_command_monitoring_events(document::view object) {
     }
 }
 
-/* TODO CXX-2138: Actually implement add_server_api below with new server_api options class.
-void add_server_api(options::server_api& sapi_opts, document::view object) {
-    if (!object["serverApi"])
-        return;
-
-    if (auto sav = object["serverApi"]["version"]) {
-        REQUIRE(sav.type() == type::k_string);
-        sapi_opts.version(sav);
-    } else {
+options::server_api create_server_api(document::view object) {
+    document::element sav;
+    if (!(sav = object["serverApi"]["version"])) {
         FAIL("must specify a version when using serverApi");
     }
 
+    REQUIRE(sav.type() == type::k_string);
+    auto version =
+        options::server_api::version_from_string(string::to_string(sav.get_string().value));
```

Comment:
Minor: Now that `version_from_string` takes a `string_view` I think this can be simplified:
```suggestion
        options::server_api::version_from_string(sav.get_string().value);
```

Reply:
Done.